src/Uno.UI/UI/Xaml/Controls/Repeater/FlowLayoutAlgorithm.cs

```diff
@@ -0,0 +1,783 @@
+					(direction == GenerateDirection.Forward && elementMajorStart < rectMajorEnd && elementMinorStart < rectMinorEnd) ||
+					(direction == GenerateDirection.Backward && elementMajorEnd > rectMajorStart && elementMinorEnd > rectMinorStart);
+			}
+
+			return shouldContinue;
+		}
+
+		Rect EstimateExtent(Size availableSize, string layoutId)
+		{
+			UIElement firstRealizedElement = null;
+			Rect firstBounds = default;
+			UIElement lastRealizedElement = null;
+			Rect lastBounds = default;
+			int firstDataIndex = -1;
+			int lastDataIndex = -1;
+
+			if (m_elementManager.GetRealizedElementCount > 0)
+			{
+				firstRealizedElement = m_elementManager.GetAt(0);
+				firstBounds = m_elementManager.GetLayoutBoundsForRealizedIndex(0);
+				firstDataIndex = m_elementManager.GetDataIndexFromRealizedRangeIndex(0);
+
+				int last = m_elementManager.GetRealizedElementCount - 1;
+				lastRealizedElement = m_elementManager.GetAt(last);
+				lastDataIndex = m_elementManager.GetDataIndexFromRealizedRangeIndex(last);
+				lastBounds = m_elementManager.GetLayoutBoundsForRealizedIndex(last);
+			}
+
+			Rect extent = m_algorithmCallbacks.Algorithm_GetExtent(
+				availableSize,
+				m_context,
+				firstRealizedElement,
+				firstDataIndex,
+				firstBounds,
+				lastRealizedElement,
+				lastDataIndex,
+				lastBounds);
+
+			REPEATER_TRACE_INFO("%*s Extent: (%.0f,%.0f,%.0f,%.0f). \n", m_context.Indent, layoutId, extent.X, extent.Y, extent.Width, extent.Height);
+			return extent;
+		}
+
+		void RaiseLineArranged()
+		{
+			var realizationRect = RealizationRect;
+			if (realizationRect.Width != 0.0f || realizationRect.Height != 0.0f)
+			{
+				int realizedElementCount = m_elementManager.GetRealizedElementCount;
+				if (realizedElementCount > 0)
+				{
+					global::System.Diagnostics.Debug.Assert(m_firstRealizedDataIndexInsideRealizationWindow != -1 && m_lastRealizedDataIndexInsideRealizationWindow != -1);
+					int countInLine = 0;
+					var previousElementBounds = m_elementManager.GetLayoutBoundsForDataIndex(m_firstRealizedDataIndexInsideRealizationWindow);
+					var currentLineOffset = MajorStart(previousElementBounds);
+					var currentLineSize = MajorSize(previousElementBounds);
+					for (int currentDataIndex = m_firstRealizedDataIndexInsideRealizationWindow; currentDataIndex <= m_lastRealizedDataIndexInsideRealizationWindow; currentDataIndex++)
+					{
+						var currentBounds = m_elementManager.GetLayoutBoundsForDataIndex(currentDataIndex);
+						if (MajorStart(currentBounds) != currentLineOffset)
+						{
+							// Staring a new line
+							m_algorithmCallbacks.Algorithm_OnLineArranged(currentDataIndex - countInLine, countInLine, currentLineSize, m_context);
+							countInLine = 0;
+							currentLineOffset = MajorStart(currentBounds);
+							currentLineSize = 0;
+						}
+
+						currentLineSize = Math.Max((float)(currentLineSize), MajorSize(currentBounds));
+						countInLine++;
+						previousElementBounds = currentBounds;
```

Comment:
Codacy found an issue: [Remove this useless assignment to local variable 'previousElementBounds'.](https://app.codacy.com/gh/unoplatform/uno/pullRequest?prid=5863680)